src/User.php

```diff
@@ -1093,12 +1093,15 @@ public function prepareInputForUpdate($input)
         }
 
        // Security on default entity  update
-        if (
-            isset($input['entities_id'])
-            && ($input['entities_id'] > 0)
-            && (!in_array($input['entities_id'], Profile_User::getUserEntities($input['id'])))
-        ) {
-            unset($input['entities_id']);
+        if (isset($input['entities_id'])) {
+            if (
+                ($input['entities_id'] > 0)
+                && (!in_array($input['entities_id'], Profile_User::getUserEntities($input['id'])))
+            ) {
+                unset($input['entities_id']);
+            } elseif ($input['entities_id'] == -1) {
+                $input['entities_id'] = 'NULL';
```

Comment:
Maybe the same should be done in the `prepareInputForAdd` method.

Reply:
No, because entities_id at creation is also used to create an authorization.